lib/Runtime/Library/SameValueComparer.h

```diff
@@ -31,39 +31,56 @@ namespace Js
             }
         }
 
+        static hash_t HashDouble(double d)
+        {
+            if (JavascriptNumber::IsNan(d))
+            {
+                return 0;
+            }
+
+            if (zero)
+            {
+                // SameValueZero treats -0 and +0 the same, so normalize to get same hash code
+                if (JavascriptNumber::IsNegZero(d))
+                {
+                    d = 0.0;
+                }
+            }
+
+            __int64 v = *(__int64*)&d;
```

Comment:
Don't we have problem here with different representation of a double of the same value here?